src/realm/query_engine.hpp

```diff
@@ -166,8 +166,8 @@ class ParentNode {
 
     double cost() const
     {
-        return 8 * bitwidth_time_unit / m_dD +
-               m_dT; // dt = 1/64 to 1. Match dist is 8 times more important than bitwidth
+        // dt = 1/64 to 1. Match dist is 8 times more important than bitwidth
+        return 8 * bitwidth_time_unit / m_dD + m_dT;
```

Comment:
I'm not confident we never set m_dD to zero anywhere (although we shouldn't be?), maybe REALM_ASSERT_DEBUG(m_dD > 0)?

Reply:
I am pretty sure it is never zero. Apart from the initial assignment, it is updated in Query::aggregate_internal alone, and these (2) calculations can not result in zero.

Reply:
ok, but given the difficulty of tracking down the calculations spread around everywhere and to prevent some future change from making this mistake, maybe it's worth guarding the division here? (not blocking)